scripts/cleanup/unusued_compilation.py

```diff
@@ -0,0 +1,136 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import fnmatch
+import os
+import shutil
+import sys
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../build'))
+import ci_build
+
+EXTENSIONS_OF_INTEREST = ('.c', '.cc', '.cpp', '.cxx', '.h', '.hpp', '.hxx', '.inl')
+EXCLUSIONS = (
+    'AZ_DECLARE_MODULE_CLASS(', 
+    'REGISTER_QT_CLASS_DESC(',
+    'TEST(',
+    'TEST_F(',
+    'TEST_P(',
+    'INSTANTIATE_TEST_CASE_P(',
+    'INSTANTIATE_TYPED_TEST_CASE_P(',
+    'AZ_UNIT_TEST_HOOK(',
+    'IMPLEMENT_TEST_EXECUTABLE_MAIN(',
+    'BENCHMARK_REGISTER_F(',
+    'DllMain(',
+    'wWinMain(',
+    'AZ_DLL_EXPORT',
+    'CreatePluginInstance('
+)
+PATH_EXCLUSIONS = (
+    '*\\Platform\\Android\\*',
+    '*\\Platform\\Common\\*',
```

Comment:
Keep in mind these are windows path separators. Did you try forward slashes with the fnmatch call?